api/src/test/java/org/apache/iceberg/variants/TestSerializedObject.java

```diff
@@ -257,13 +246,40 @@ public void testLargeObject(boolean sortFieldNames) {
     assertThat(object.type()).isEqualTo(PhysicalType.OBJECT);
     assertThat(object.numFields()).isEqualTo(10_000);
 
-    for (Map.Entry<String, SerializedPrimitive> entry : fields.entrySet()) {
+    for (Map.Entry<String, VariantPrimitive<?>> entry : fields.entrySet()) {
       VariantValue fieldValue = object.get(entry.getKey());
-      assertThat(fieldValue.type()).isEqualTo(PhysicalType.STRING);
-      assertThat(fieldValue.asPrimitive().get()).isEqualTo(entry.getValue().get());
+      assertThat(fieldValue.getClass()).isEqualTo(SerializedPrimitive.class);
+      assertThat(fieldValue.asPrimitive().sizeInBytes()).isEqualTo(5 + 10);
```

Comment:
I think we should revert this change. This is to test out the object has many fields. I don't think we need to check the size here.